src/asm/statements.rs

```diff
@@ -52,7 +52,20 @@ impl std::fmt::Display for Instruction<'_> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         write!(f, "{}", color!(self.op, OwoColorize::bright_blue))?;
         if let Some(args) = self.args {
-            write!(f, " {}", demangle::contents(args, f.alternate()))?;
+            let args = demangle::contents(args, f.alternate());
+            let args = args.as_ref();
```

Comment:
I think it might be easier to do the `replace_all` thing from regex, similar to how names are demangled, performance wise should be similar but with no loops anywhere

Reply:
```diff
diff --git a/src/asm/statements.rs b/src/asm/statements.rs
index 109f67c..3448005 100644
--- a/src/asm/statements.rs
+++ b/src/asm/statements.rs
@@ -53,19 +53,7 @@ impl std::fmt::Display for Instruction<'_> {
         write!(f, "{}", color!(self.op, OwoColorize::bright_blue))?;
         if let Some(args) = self.args {
             let args = demangle::contents(args, f.alternate());
-            let args = args.as_ref();
-            let mut prev = 0;
-            f.write_str(" ")?;
-
-            for label in demangle::local_labels(args) {
-                // write all before the first label and between each label
-                f.write_str(&args[prev..label.start()])?;
-                write!(f, "{}", color!(label.as_str(), OwoColorize::bright_black))?;
-                prev = label.end();
-            }
-
-            // write all remaining arguments
-            f.write_str(&args[prev..])?;
+            write!(f, " {}", demangle::labels(&args))?;
         }
         Ok(())
     }
diff --git a/src/demangle.rs b/src/demangle.rs
index edb7fef..1929358 100644
--- a/src/demangle.rs
+++ b/src/demangle.rs
@@ -40,6 +40,18 @@ pub fn local_labels(input: &str) -> regex::Matches {
     LOCAL_LABELS.find_iter(input)
 }
 
+struct Labelcol;
+impl Replacer for Labelcol {
+    fn replace_append(&mut self, caps: &regex::Captures<'_>, dst: &mut String) {
+        use std::fmt::Write;
+        write!(dst, "{}", color!(&caps[0], OwoColorize::bright_black)).unwrap();
+    }
+}
+
+pub fn labels(input: &str) -> Cow<'_, str> {
+    LOCAL_LABELS.replace_all(input, Labelcol)
+}
+
 struct Demangler {
     full_name: bool,
 }
```

Reply:
Oh, cool! I wasn't aware of the `Replacer` trait in `regex`. I'll add this in.